### ISSUE TITLE
Add FMix preprocessing layer

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#254]
Paper: [FMix: Enhancing Mixed Sample Data Augmentation](https://arxiv.org/abs/2002.12047)
Code: https://github.com/ecs-vlc/FMix

(both tensorflow and pytorch implementation are provided).

![fmix_example](https://user-images.githubusercontent.com/17668390/161284341-7bfb24fb-330a-4c6c-83d5-efcf0de20414.png)

![fmix_3d](https://user-images.githubusercontent.com/17668390/161284368-99d99bf1-6e7c-4d91-b6f3-cb5605e0f63d.gif)

**Comment by LukeWood** [comment]
That looks pretty cool.  So it is like CutMix but with much more organic looking shapes?

**Comment by innat** [comment]
yes, it's just a variant of cutmix/mixup. It uses masks sampled from Fourier space to mix training examples.

**Comment by LukeWood** [comment]
Thats so cool.  @innat would you like to contribute this or should I find a contributor?  I have someone in mind that might be a great match for this if you aren't hoping to add it. 

cheers

**Comment by innat** [comment]
@LukeWood I would love to. But I'm busy with some other essentials. Please find some contributors.

**Comment by quantumalaviya** [comment]
I can work on this @LukeWood

**Comment by LukeWood** [comment]
Awesome, feel free to begin a PR and send for review when needed.
